crash on line staging

```
panic: PanicInfo { payload: Any, message: Some(index out of bounds: the len is 2 but the index is 2), location: Location { file: "asyncgit/src/sync/staging/mod.rs", line: 44, col: 25 } }
trace:
   0: backtrace::capture::Backtrace::new
   1: gitui::set_panic_handlers::{{closure}}
   2: std::panicking::rust_panic_with_hook
   3: std::panicking::begin_panic_handler::{{closure}}
   4: std::sys_common::backtrace::__rust_end_short_backtrace
   5: _rust_begin_unwind
   6: core::panicking::panic_fmt
   7: core::panicking::panic_bounds_check
   8: asyncgit::sync::staging::NewFromOldContent::add_old_line
   9: asyncgit::sync::staging::apply_selection
  10: <gitui::components::diff::DiffComponent as gitui::components::Component>::event
  11: gitui::components::event_pump
  12: <gitui::tabs::status::Status as gitui::components::Component>::event
  13: gitui::components::event_pump
  14: gitui::main
  15: std::sys_common::backtrace::__rust_begin_short_backtrace
  16: _main
```

<img width="423" alt="image" src="https://user-images.githubusercontent.com/776816/112617219-f5e00c00-8e24-11eb-8916-a241a370af53.png">

--- a/asyncgit/src/sync/staging/mod.rs
+++ b/asyncgit/src/sync/staging/mod.rs
@@ -121,7 +121,9 @@ pub(crate) fn apply_selection(
                         .trim()
                 );
 
-                if hunk_line.origin() == '<' {
+                if hunk_line.origin() == '<'
+                    || hunk_line.origin() == '>'
+                {
                     break;
                 }
 

--- a/asyncgit/src/sync/staging/stage_tracked.rs
+++ b/asyncgit/src/sync/staging/stage_tracked.rs
@@ -108,6 +108,49 @@ mod test {
     }
 
     #[test]
+    fn test_panic_stage_no_newline() {
+        static FILE_1: &str = r"a = 1
+b = 2";
+
+        static FILE_2: &str = r"a = 2
+b = 3
+c = 4";
+
+        let (path, repo) = repo_init().unwrap();
+        let path = path.path().to_str().unwrap();
+
+        write_commit_file(&repo, "test.txt", FILE_1, "c1");
+
+        repo_write_file(&repo, "test.txt", FILE_2).unwrap();
+
+        stage_lines(
+            path,
+            "test.txt",
+            false,
+            &[
+                DiffLinePosition {
+                    old_lineno: Some(1),
+                    new_lineno: None,
+                },
+                DiffLinePosition {
+                    old_lineno: Some(2),
+                    new_lineno: None,
+                },
+            ],
+        )
+        .unwrap();
+
+        let diff =
+            get_diff(path, String::from("test.txt"), true).unwrap();
+
+        assert_eq!(diff.lines, 5);
+        assert_eq!(
+            diff.hunks[0].lines[0].content,
+            String::from("@@ -1,2 +1 @@\n")
+        );
+    }
+
+    #[test]
     fn test_unstage() {
         static FILE_1: &str = r"0
 ";

